app/controllers/admin_controller.rb

```diff
@@ -389,4 +331,43 @@ def smtp_test
     s.close
   end
 
+  def moderator_or_admin_user
```

Comment:
Should these maybe all go under a `private` section so they are only call-able from within this file?